framework/src/utils/SlepcSupport.C

```diff
@@ -0,0 +1,242 @@
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef SLEPCSUPPORT_H
+#define SLEPCSUPPORT_H
+
+#include "libmesh/libmesh_config.h"
+
+#ifdef LIBMESH_HAVE_SLEPC
+
+#include "SlepcSupport.h"
+// MOOSE includes
+#include "MultiMooseEnum.h"
+#include "InputParameters.h"
+#include "EigenProblem.h"
+#include "Conversion.h"
+
+namespace Moose
+{
+namespace SlepcSupport
+{
+
+  InputParameters
+  getSlepcValidParams()
+  {
+    InputParameters params = emptyInputParameters();
+
+    MooseEnum eigen_solve_type("POWER ARNOLDI KRYLOVSCHUR JACOBI_DAVIDSON");
+    params.addParam<MooseEnum>("eigen_solve_type", eigen_solve_type,
+                               "POWER: Power / Inverse / RQI "
+                               "ARNOLDI: Arnoldi "
+                               "KRYLOVSCHUR: Krylov-Schur "
+                               "JACOBI_DAVIDSON: Jacobi-Davidson ");
+   return params;
+  }
+
+  InputParameters
+  getSlepcEigenProblemValidParams()
+  {
+    InputParameters params = emptyInputParameters();
+
+    // We are solving a Non-Hermitian eigenvalue problem by default
+    MooseEnum eigen_problem_type("HERMITIAN NON_HERMITIAN GEN_HERMITIAN GEN_NON_HERMITIAN GEN_INDEFINITE POS_GEN_NON_HERMITIAN", "NON_HERMITIAN");
+    params.addParam<MooseEnum>("eigen_problem_type", eigen_problem_type,
+                               "HERMITIAN: Hermitian "
+                               "NON_HERMITIAN: Non-Hermitian "
+                               "GEN_HERMITIAN: Gerneralized Hermitian "
+                               "GEN_NON_HERMITIAN: Generalized Non-Hermitian "
+                               "GEN_INDEFINITE: Generalized indefinite Hermitian "
+                               "POS_GEN_NON_HERMITIAN: Generalized Non-Hermitian with positive (semi-)definite B");
+
+
+    // Which eigenvalues are we interested in
+    MooseEnum which_eigen_pairs("LARGEST_MAGNITUDE SMALLEST_MAGNITUDE LARGEST_REAL SMALLEST_REAL LARGEST_IMAGINARY SMALLEST_IMAGINARY TARGET_MAGNITUDE TARGET_REAL TARGET_IMAGINARY ALL_EIGENVALUES", "SMALLEST_MAGNITUDE");
+    params.addParam<MooseEnum>("which_eigen_pairs", which_eigen_pairs,
+                               "LARGEST_MAGNITUDE: largest magnitude "
+                               "SMALLEST_MAGNITUDE: smallest magnitude "
+                               "LARGEST_REAL: largest real "
+                               "SMALLEST_REAL: smallest real "
+                               "LARGEST_IMAGINARY: largest imaginary "
+                               "SMALLEST_IMAGINARY: smallest imaginary "
+                               "TARGET_MAGNITUDE: target magnitude "
+                               "TARGET_REAL: target real "
+                               "TARGET_IMAGINARY: target imaginary "
+                               "ALL_EIGENVALUES: all eigenvalues ");
```

Comment:
The doco string should read something like: "Which eigenvalue pair to obtain from the solution". The values in the `MooseEnum` are quite self-explanatory now...